wal/wal.go

```diff
@@ -532,21 +540,28 @@ func (w *WAL) saveState(s *raftpb.HardState) error {
 }
 
 func (w *WAL) Save(st raftpb.HardState, ents []raftpb.Entry) error {
-	w.mu.Lock()
```

Comment:
I agree with @heyitsanthony. Let's just revert this change for now? And keep the change for SaveSnapshot?